modules/core/src/main/java/org/locationtech/jts/operation/distance/DistanceOp.java

```diff
@@ -428,7 +428,7 @@ private void computeMinDistance(LineString line, Point pt,
         if (dist < minDistance) {
           minDistance = dist;
           LineSegment seg = new LineSegment(coord0[i], coord0[i + 1]);
-          Coordinate segClosestPoint = seg.closestPoint(coord);
+          Coordinate segClosestPoint = seg.closestPoint(coord, i > 0);
```

Comment:
If you're looking to scavenge performance, perhaps this functionality should be provided as a static function?  Probably on `LineSegment`.  I've been thinking that many of the LineSegment methods could also be provided as static functions, for one-off use cases.

It really needs some performance testing, though - too easy to prematurely optimize.

Reply:
If you have 10 LineSegments, each getting closer to the point, then the current code does 10 point-to-line and 11+10=19 point-to-point calculations. Each intermediate point is calculated twice. There should only be 10 point-to-line and 11 point-to-point calculations.

Pertinent to your other comment, I agree the skip0 boolean is an obscure solution. Better for the DistanceOp to explicitly do the point-to-line and point-to-point calculations separately, with point-to-line being of the form orthgonalPoint-to-line.

And static functions also make sense. You don't need to instantiate a LineSegment just to do a calculation then throw it away. Bad for the garbage-collector. Now if LineSegments and their attributes were declared final, then maybe they would be comparable, but that isn't currently an option.

The code below is problematic as well, because it first does the calculation without instances in Distance.pointToSegment() and then repeats the calculation **with** instances seg.closestPoint. If you separate point-to-point and orthogonalPoint-to-line, I think you can eliminate the duplication.

double dist = Distance.pointToSegment( coord, coord0[i], coord0[i + 1] );
if (dist < minDistance) {
   minDistance = dist;
   LineSegment seg = new LineSegment(coord0[i], coord0[i + 1]);
   Coordinate segClosestPoint = seg.closestPoint(coord); **//This is the same calculation as Distance.pointToSegment!**
}